### PR TITLE
Explicitly note coercion to string for .contains

### PR DESCRIPTION
**Pull request by VynceMontgomery**
as per #992

### CODE REVIEW

File: doc/Type/Cool.pod6 [hunk 1]
```diff
@@ -1163,12 +1163,18 @@ L<Regex|/type/Regex> as a list of strings.
...
+to L<Str|/type/Str>, and searches for C<$needle> in the string starting from C<$start>.
 Returns C<True> if the invocant contains the C<$needle> at any position within
 the string. If $start is provided skip as many characters.
 
     say "Hello, World".contains('hello');      # OUTPUT: «False␤»
     say "Hello, World".contains(',');          # OUTPUT: «True␤»
 
+Note that because of how a L<List|/type/List> or L<Array|/type/Array> is 
+L<coerced|/type/List#method_Str> into a L<Str|/type/Str>, the results may 
+sometimes be surprising. See L<traps|/language/traps>.
```
Comment (AlexDaniel) [inline doc/Type/Cool.pod6]: uh, is it possible to give a more precise link?

Comment (toolforger) [inline doc/Type/Cool.pod6]: +1 : It would be nice to post an anchored link, i.e. /language/traps#coercion-with-lists or whatever the anchor on that page might be.

File: doc/Type/Cool.pod6 [hunk 2]
```diff
@@ -1163,12 +1163,18 @@ L<Regex|/type/Regex> as a list of strings.
 
     multi method contains(Cool:D: Str(Cool) $needle, Cool $start? --> Bool:D)
 
+Coerces the first two arguments (in method form, also counting the invocant)
+to L<Str|/type/Str>, and searches for C<$needle> in the string starting from C<$start>.
 Returns C<True> if the invocant contains the C<$needle> at any position within
 the string. If $start is provided skip as many characters.
 
     say "Hello, World".contains('hello');      # OUTPUT: «False␤»
```
Comment (toolforger) [inline doc/Type/Cool.pod6]: I'm wondering what the output for `.contains('Hello")` would be.
(Sorry if that is out of scope of the PR.)

File: doc/Type/Cool.pod6 [hunk 3]
```diff
@@ -1163,12 +1163,18 @@ L<Regex|/type/Regex> as a list of strings.
 
     multi method contains(Cool:D: Str(Cool) $needle, Cool $start? --> Bool:D)
 
+Coerces the first two arguments (in method form, also counting the invocant)
```
Comment (toolforger) [inline doc/Type/Cool.pod6]: This should be `Coerces the invocant and $needle`. Not everybody knows by heart what exactly "method form" means.
Is "invocant" the right terminology in Perl6? I would have expected that to be "target", but then that might be the terminology from elsewhere.

File: doc/Type/Cool.pod6 (hunk 1 above)
Comment (VynceMontgomery) [inline doc/Type/Cool.pod6]: Yeah, at the time of writing, I didn't know what that anchor would look like.  Now that Cool Tricks sextion of traps has been added, I can update the link here.

File: doc/Type/Cool.pod6 (hunk 2 above)
Comment (VynceMontgomery) [inline doc/Type/Cool.pod6]: Yeah, out of scope, but I'm happy to add more examples. I'm not sure how many is too many.

File: doc/Type/Cool.pod6 (hunk 3 above)
Comment (VynceMontgomery) [inline doc/Type/Cool.pod6]: I copied from a nearby source. As far as I can tell, it's not yet explained well enough that I can be sure what it means, so I've got to go a little cargo cult and trust that the other answers are correct.

**Comment by coke** [comment]
This PR fails 'make xtest' (but is easily fixable)

```
xt/trailing-whitespace.t (Wstat: 256 Tests: 376 Failed: 1)
  Failed test:  94
  Non-zero exit status: 1
```

**Comment by coke** [comment]
The sentence you add overlaps with some of the text that's already there, for contains. (regarding start and needle)

**Comment by VynceMontgomery** [comment]
Really? Trailing whitespace is not allowed in doc text? OK. I've removed it where I found it.

**Comment by toolforger** [comment]
> Really? Trailing whitespace is not allowed in doc text?

Trailing whitespace is invisible, and disallowing it removes corner 
cases from tools that process the text.
That's the usual reason why such a policy is considered reasonable.

**Comment by zoffixznet** [comment]
Thanks.
